src/Lib/Canvas.vala

```diff
@@ -262,12 +307,16 @@ public class Akira.Lib.Canvas : Goo.Canvas {
     private void on_request_change_cursor (Gdk.CursorType? cursor_type) {
         if (cursor_type == null) {
             set_cursor_by_edit_mode ();
-        } else {
-            set_cursor (cursor_type);
+            return;
         }
+
+        set_cursor (cursor_type);
     }
 
     private void set_cursor (Gdk.CursorType? cursor_type) {
+        // debug (@"Setting cursor: $cursor_type");
+        current_cursor = cursor_type;
```

Comment:
Move this at the end of the function as we don't want to change the current_cursor until the GdkCursor was updated.

Reply:
Ok